### DISCUSSION TITLE
Can flux kustomization dry-run be performed in cmdline?

### DISCUSSION THREAD

**Discussion by maplewf** [fluxcd/flux2#5084] [General] [answered]
Hi,

As I know, flux kustomization controller will execute dry-run before really applying manifests, this is very nice since it can prevent malformed configuration is applied. But I wonder if this dry-run check can be executed with cmdline, since I want to integrate this great check with my git source code workflow. 

I know there is `flux build kustomization` for building/checking manifest against with cluster or offline, but the check result of this cmdline doesn't reach my expectation. For example, if I add a manifest with kind `A`, but CRD for `A` is missing from cluster, then dry-run from flux kustomization controller can detect it but `flux build kustomization` can't.

So is there way for us to do same check in source code as flux kustomization controller dry-run did? Thanks

**Comment by stefanprodan** [comment, marked as answer]
Yes the dry-run is implemented in the CLI with `flux diff kustomization` which runs against the live cluster without modifying its state, the diff is what the controller does before the actual server-side apply.

**Reply by maplewf** [reply]
@stefanprodan This is exaxtly what I want, thanks
